src/ucp/wireup/address.h

```diff
@@ -294,4 +300,31 @@ uint8_t ucp_address_is_am_only(const void *address);
  */
 size_t ucp_address_iface_seg_size(const uct_iface_attr_t *iface_attr);
 
+static inline uint8_t ucp_address_packed_release_version(uint8_t api_minor)
+{
+    return api_minor % UCP_ADDRESS_RELEASE_VERSION_MAX;
+}
+
+static inline uint8_t ucp_address_pack_release_version()
+{
+    return ucp_address_packed_release_version(UCP_API_MINOR);
+}
+
+/* Release version is packed into 4 bits. Unpack it assuming that we do not
+ * support wire compatibility for more than 8 versions (therefore we take the
+ * value which is closest to our release version)
+ */
+static inline unsigned
+ucp_address_unpack_release_version(uint8_t packed_version)
```

Comment:
IMO we can keep these slow-path functions in C file, and expose only ucp_address_packed_release_version and ucp_address_unpack_release_version prototypes in the H file for the gtest
Also  UCP_ADDRESS_RELEASE_VERSION_BITS, UCP_ADDRESS_RELEASE_VERSION_MAX could be moved back to C file
This would also be better encapsulation and hiding impl details of the address packing